portfolio/src/main/webapp/script.js

```diff
@@ -139,6 +141,32 @@ function formatDate(epochDate) {
 
 }
 
+/**
+ * Deletes all guestbook comments from the website if the 
+ * correct password is entered.
+ */
+function deleteData() {
+  const enteredPassword = document.getElementById('pwd').value;
+  const requestURL = '/delete-data?password=' + enteredPassword;
+  const request = new Request(requestURL, {method: 'POST'});
+  
+  fetch(request).then((response) => {
+    const passwordMessage = document.getElementById("password-fail");
+    if (response.status === 403) {
```

Comment:
Instead of this just check for ok status and if it is not ok, check if error_message is set or not. If error message is not set then you can set a default error message like "Unknown error occurred". then you can display error code and error message in innertext. This does not make any assumption about the backend except that it might return an error_message.